### ISSUE TITLE
[Serve] `serve start` and `serve run` don't use HTTP

### ISSUE THREAD

**Issue by shrekris-anyscale** [ray-project/ray#26308]
### Description

It would be nice to add that `serve start` and `serve run` don't use HTTP somewhere in the docs. Since all the other `serve` CLI commands use HTTP, it's surprising that these two use Ray client. This can lead to confusing behavior when working with long-lived, remote Ray clusters.

### Link

https://docs.ray.io/en/master/serve/production.html

**Comment by gariepyalex** [comment]
This page also uses the Ray client address instead of the HTTP API of the Ray dashboard: https://docs.ray.io/en/latest/serve/dev-workflow.html#testing-on-a-remote-cluster

Using the client URL doesn't work as documented and gives a runtime error

```
ValueError: Got an unexpected Ray client address "ray://localhost:10001" while trying to connect to the Ray dashboard. The dashboard SDK requires the Ray dashboard server's HTTP(S) address (which should start with "http://" or "https://", not "ray://"). If this address wasn't passed explicitly, it may be set in the RAY_ADDRESS environment variable.
```

**Comment by shrekris-anyscale** [comment]
@gariepyalex Thanks for letting us know! That error's a bit surprising– since [`serve run` uses Ray client](https://github.com/ray-project/ray/blob/80749160e133199539d67848d58f4935683f97bb/python/ray/serve/scripts.py#L305), I would expect a Ray client address to be required there.

Could you confirm that you noticed this error with `serve run` (and not `serve deploy`)?

**Comment by gariepyalex** [comment]
You are right, I entered the wrong command and got the error with `serve deploy`. False alarm!
